pkg/codegen/internal/test/program_driver.go

```diff
@@ -19,78 +19,147 @@ import (
+		// Blocked on go:
+		// https://github.com/pulumi/pulumi-aws/issues/1632
+		//
+		// Blocked on nodejs
+		// Starting with:
+		// aws-eks.ts:34:65 - error TS1005: ';' expected.
+		//
+		// 34     for (const range of zones.names.map((k, v) => {key: k, value: v})) {
+		//                                                                    ~
 	},
 	{
-		ProgramFile: "aws-fargate",
+		Name:        "aws-fargate",
 		Description: "AWS Fargate",
+		SkipCompile: codegen.NewStringSet("go"),
+		// Blocked on go:
+		// https://github.com/pulumi/pulumi-aws/issues/1632
 	},
 	{
-		ProgramFile: "aws-s3-logging",
+		Name:        "aws-s3-logging",
 		Description: "AWS S3 with logging",
+		SkipCompile: codegen.NewStringSet("dotnet", "nodejs"),
+		// Blocked on dotnet:
+		// /codegen/internal/test/testdata/aws-s3-logging-pp/aws-s3-logging.cs(21,71):
+		// error CS0023: Operator '?' cannot be applied to operand of type 'ImmutableArray<BucketLogging>'
+		//
+		// Blocked on nodejs:
+		// It looks like this is being parsed as a ternary expression
+		// aws-s3-logging.ts:8:89 - error TS1005: ':' expected.
+		//
+		// 8: export const targetBucket = bucket.loggings.apply(loggings => loggings?[0]?.targetBucket);
+		//                                                                                            ~
 	},
 	{
-		ProgramFile: "aws-webserver",
+		Name:        "aws-webserver",
 		Description: "AWS Webserver",
+		SkipCompile: codegen.NewStringSet("go"),
+		// Blocked on go:
+		// https://github.com/pulumi/pulumi-aws/issues/1632
 	},
 	{
-		ProgramFile: "azure-native",
+		Name:        "azure-native",
 		Description: "Azure Native",
-		Skip:        codegen.NewStringSet("go"),
+		Skip:        codegen.NewStringSet("go", "nodejs"),
+		// Blocked on go:
+		// Blocked on nodjs:
+		// Types do not line up
 	},
 	{
-		ProgramFile: "azure-sa",
+		Name:        "azure-sa",
 		Description: "Azure SA",
 	},
 	{
-		ProgramFile: "kubernetes-operator",
+		Name:        "kubernetes-operator",
 		Description: "K8s Operator",
 	},
 	{
-		ProgramFile: "kubernetes-pod",
+		Name:        "kubernetes-pod",
 		Description: "K8s Pod",
+		SkipCompile: codegen.NewStringSet("go", "nodejs"),
+		// Blocked on go:
+		// Blocked on nodejs:
+		// Types do not line up
 	},
 	{
-		ProgramFile: "kubernetes-template",
+		Name:        "kubernetes-template",
 		Description: "K8s Template",
 	},
 	{
-		ProgramFile: "random-pet",
+		Name:        "random-pet",
 		Description: "Random Pet",
 	},
 	{
-		ProgramFile: "resource-options",
+		Name:        "aws-resource-options",
 		Description: "Resource Options",
+		SkipCompile: codegen.NewStringSet("go"),
+		// Blocked on go:
+		// generating invalid aws.Provider code
 	},
 	{
-		ProgramFile: "secret",
+		Name:        "aws-secret",
 		Description: "Secret",
 	},
 	{
-		ProgramFile: "functions",
+		Name:        "functions",
 		Description: "Functions",
+		SkipCompile: codegen.NewStringSet("go", "dotnet"),
+		// Blocked on go:
```

Comment:
This looks very unfortunate, worth a new issue?

Reply:
My plan is to do a pass after this merges and file an issue for each case that doesn't compile.

Reply:
New issues are here: https://github.com/pulumi/pulumi/issues/7804#issuecomment-928135309